nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-headless-server/src/main/java/org/apache/nifi/headless/HeadlessNiFiServer.java

```diff
@@ -0,0 +1,199 @@
+            final String sensitivePropProviderVal = props.getProperty(StringEncryptor.NF_SENSITIVE_PROPS_PROVIDER);
+            final String sensitivePropValueNifiPropVar = props.getProperty(StringEncryptor.NF_SENSITIVE_PROPS_KEY, DEFAULT_SENSITIVE_PROPS_KEY);
+
+            StringEncryptor encryptor = StringEncryptor.createEncryptor(sensitivePropAlgorithmVal, sensitivePropProviderVal, sensitivePropValueNifiPropVar);
+            VariableRegistry variableRegistry = new FileBasedVariableRegistry(props.getVariableRegistryPropertiesPaths());
+            BulletinRepository bulletinRepository = new VolatileBulletinRepository();
+            StandardFlowRegistryClient flowRegistryClient = new StandardFlowRegistryClient();
+            flowRegistryClient.setProperties(props);
+
+            FlowController flowController = FlowController.createStandaloneInstance(
+                    flowFileEventRepository,
+                    props,
+                    authorizer,
+                    auditService,
+                    encryptor,
+                    bulletinRepository,
+                    variableRegistry,
+                    flowRegistryClient,
+                    extensionManager
+                    );
+
+            flowService = StandardFlowService.createStandaloneInstance(
+                    flowController,
+                    props,
+                    encryptor,
+                    null, // revision manager
+                    authorizer);
+
+            // start and load the flow
+            flowService.start();
+            flowService.load(null);
+            flowController.onFlowInitialized(true);
+            FlowManager flowManager = flowController.getFlowManager();
+            flowManager.getGroup(flowManager.getRootGroupId()).startProcessing();
+
+            logger.info("Flow loaded successfully.");
+        } catch (Exception e) {
+            // ensure the flow service is terminated
+            if (flowService != null && flowService.isRunning()) {
+                flowService.stop(false);
+            }
+            startUpFailure(new Exception("Unable to load flow due to: " + e, e));
+        }
+    }
+
+    private void startUpFailure(Throwable t) {
+        System.err.println("Failed to start flow service: " + t.getMessage());
+        System.err.println("Shutting down...");
+        logger.warn("Failed to start headless server... shutting down.", t);
+        System.exit(1);
+    }
+
+    @Override
+    public void initialize(NiFiProperties properties, Bundle systemBundle, Set<Bundle> bundles, ExtensionMapping extensionMapping) {
+        this.props = properties;
+        this.systemBundle = systemBundle;
+        this.bundles = bundles;
+    }
+
+    public DiagnosticsFactory getDiagnosticsFactory() {
+        return null;
+    }
+
+    public DiagnosticsFactory getThreadDumpFactory() {
+        return null;
+    }
```

Comment:
I don't really know what is involved in providing these, but just wanted to double check that we really want them to be null since then I think you can't call the nifi.sh diagnostics command without them.

Reply:
I updated the PR to use the same diagnostics stuff that the JettyServer stuff uses, just had to create it explicitly (JettyServer gets its instance injected via Spring).